rotkehlchen/tests/api/test_user_assets.py

```diff
@@ -504,7 +520,9 @@ def test_deleting_user_assets(rotkehlchen_api_server, globaldb, cache_coinlist):
     assert_proper_response(response)
     assert globaldb.get_asset_data(identifier=user_asset1_id, form_with_incomplete_data=False) is None  # noqa: E501
     # Also check that `forked` was updated
-    assert globaldb.get_asset_data(identifier=user_asset2_id, form_with_incomplete_data=False).forked is None  # noqa: E501
+
+    if globaldb.get_asset_data(identifier=user_asset2_id, form_with_incomplete_data=False) is not None:  # noqa: E501
+        assert globaldb.get_asset_data(identifier=user_asset2_id, form_with_incomplete_data=False).forked is None  # type: ignore # mypy complains about None of Assetdata | None even after checking in the previous line. # noqa: E501
```

Comment:
It can't see it since you don't save it in a variable and there is no concept of a pure function. So there is no guarantee the function will always return the same value for the same input.

So this should probably work.

```suggestion
    assert (asset_data := globaldb.get_asset_data(identifier=user_asset2_id, form_with_incomplete_data=False)) is not None
    assert asset_data.forked is None
```